apt/preferences.sls

```diff
@@ -7,13 +7,15 @@
 {% set default_url = apt.get('default_url', apt_map.default_url) %}
 
 /etc/apt/preferences:
-  {% if remove_preferences %}
-  file.absent
-  {% else %}
   file.managed:
     - mode: '0644'
     - user: root
     - group: root
+  {% if remove_preferences %}
+    - contents: ''
+    - contents_newline: False
+  {% else %}
```

Comment:
> Now, if we want to get coherent, yeah, having the remove_preferences and remove_sources_list act the same way, having the remove place it to empty file, and if not have the replace: False to avoid warning...

@ze42 yes, the modification I suggested is to have a coherent behaviour in the formula. Notice that with the suggested changes (as is happening with the `/etc/apt/sources.list`), the `replace:` attribute is not needed, as the zeroing `contents:...` lines are added only if `remove_preferences: True`, Otherwise, the resource will just set permissions on the file (which will produce no warnings)

wdyt? Am I missing something?

Reply:
When we clean, we had file.absent for preferences, now we clean it up with specified contents.

When we do no remove preferences, we set permissions, but without any contents/sources, is exactly the case where I did get my warning in the first place... That's why I suggest explicitly adding the replace: False in those cases on both files.

Reply:
You're right. I *was* missing something. OK, I get your point.